libsolidity/experimental/ast/TypeSystemHelper.cpp

```diff
@@ -356,14 +356,14 @@ std::string formatSortSuffix(Sort const& _sort, TypeEnvironment const& env)
 
 }
 
-std::string TypeEnvironmentHelpers::typeToString(Type const& _type) const
+std::string TypeEnvironmentHelpers::typeToString(Type const& _type, bool _resolve) const
```

Comment:
This does not make sense either. Unresolved types do not make sense, so this should *always* resolve.

Reply:
It's never really reasonable to look at unresolved types, even for debugging :-).

Reply:
I really don't think it's unreasonable. It helped me multiple times already :)

Reply:
I'd bet that was only due to the bug in resolving types, though. In general, looking at unresolved types is looking at a random previous value of a variable that's long overridden without even having any meaningful notion of it being a reasonable intermediate step. It's significantly more confusing to have this option here suggesting that an unresolved type provides any valuable information at all.

Reply:
Yeah, but I don't think these kinds of bugs will be that rare.

But anyway. This is now removed along with other changes. I'll leave it at that for now, but be sure that you'll see a PR with it again as soon as I encounter this kind of problem once more :)